### PR TITLE
Doc comment corrections

### PR DESCRIPTION
**Pull request by amartini51**
Collect several documentation comment fixes.

Resolves:

- <rdar://problem/72592923> (SR-13983)
- <rdar://problem/70543419>
- <rdar://problem/58716408>
- <rdar://problem/65609221>
- <rdar://problem/51409021>

### CODE REVIEW

**Comment by amartini51** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/FloatingPoint.swift [hunk 1]
```diff
@@ -1157,7 +1157,7 @@ public protocol FloatingPoint: SignedNumeric, Strideable, Hashable
 
   /// A Boolean value indicating whether the instance is infinite.
   ///
-  /// Note that `isFinite` and `isInfinite` do not form a dichotomy, because
+  /// Note that `isFinite` and `isInfinite` don't form a dichotomy, because
```
Comment (tbkka) [inline stdlib/public/core/FloatingPoint.swift:1160]: "form a dichotomy"??   We should really just shorten this to:

> Note that if `x` is `NaN`, then `x.isFinite` and `x.isInfinite` are both `false`.

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 2]
```diff
@@ -92,6 +92,12 @@ public struct ${Self} {
 ${Availability(bits)}
 extension ${Self}: CustomStringConvertible {
   /// A textual representation of the value.
+  ///
+  /// For all finite values, the value of this property is a string that can be
```
Comment (tbkka) [inline stdlib/public/core/FloatingPointTypes.swift.gyb:96]: Suggest:
> For any finite value, this property provides a string ...

**Review by tbkka — APPROVED** [review]

File: stdlib/public/core/FloatingPointTypes.swift.gyb (hunk 2 above)
Comment (amartini51) [reply stdlib/public/core/FloatingPointTypes.swift.gyb]: Sure — changed in 28ad84b4145a1a64170b667ea84de4cc8f61e059.

File: stdlib/public/core/FloatingPoint.swift (hunk 1 above)
Comment (amartini51) [reply stdlib/public/core/FloatingPoint.swift]: Changed in df8f551d7f4ba96823416bf8d179aaab7d0cb529, trying an even simpler wording and adding this note under both `isFinite` and `isInfinite`.

**Review by rxwei — APPROVED** [review]

File: stdlib/public/core/FloatingPointTypes.swift.gyb [hunk 3]
```diff
@@ -108,6 +114,9 @@ extension ${Self}: CustomStringConvertible {
 ${Availability(bits)}
 extension ${Self}: CustomDebugStringConvertible {
   /// A textual representation of the value, suitable for debugging.
+  ///
+  /// This property has the same value as the `description` propyrty, except
```
Comment (kernelpop) [inline stdlib/public/core/FloatingPointTypes.swift.gyb]: *property

Comment (amartini51) [reply stdlib/public/core/FloatingPointTypes.swift.gyb]: That's an outdated diff — this spelling error was already fixed in commit b92a5bd290a3f0cbbe80028aa9bb6bbe67973209.

**Comment by amartini51** [comment]
@swift-ci please smoke test linux
